Tests/AsyncKitTests/ConnectionPoolTests.swift

```diff
@@ -13,92 +13,98 @@ final class ConnectionPoolTests: XCTestCase {
+        let connD = ManagedAtomic<FooConnection?>(nil)
+        futureD.whenSuccess { connD.store($0, ordering: .relaxed) }
+        XCTAssertNil(connD.load(ordering: .relaxed))
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 2)
-        
+
         // this time, close the connection before releasing it
-        try connC!.close().wait()
-        pool.releaseConnection(connC!)
-        XCTAssert(connD !== connB)
-        XCTAssertEqual(connD?.isClosed, false)
+        try connCRet.close().wait()
+        pool.releaseConnection(connC.load(ordering: .relaxed)!)
+        let connDRet = try futureD.wait()
+        XCTAssert(connD.load(ordering: .relaxed) !== connB)
+        XCTAssert(connDRet === connD.load(ordering: .relaxed))
+        XCTAssertEqual(connD.load(ordering: .relaxed)?.isClosed, false)
         XCTAssertEqual(foo.connectionsCreated.load(ordering: .relaxed), 3)
     }
 
-    func testConnectionPrunning() async throws {
+    func testConnectionPruning() async throws {
         let foo = FooDatabase()
         let pool = EventLoopConnectionPool(
             source: foo,
             maxConnections: 5,
-            on: MultiThreadedEventLoopGroup(numberOfThreads: 1).next(),
-            pruneInterval: 0.1,
-            maxIdleTimeBeforePrunning: 0.15
+            pruneInterval: .milliseconds(100),
+            maxIdleTimeBeforePruning: .milliseconds(150),
+            on: MultiThreadedEventLoopGroup(numberOfThreads: 1).next()
         )
-        
+
         defer { try! pool.close().wait() }
 
-        let connA = try pool.requestConnection().wait()
-        
-        let anotherConnection1 = try pool.requestConnection().wait()
-        let anotherConnection2 = try pool.requestConnection().wait()
+        let connA = try await pool.requestConnection().get()
+
+        let anotherConnection1 = try await pool.requestConnection().get()
+        let anotherConnection2 = try await pool.requestConnection().get()
 
         pool.releaseConnection(connA)
 
-        let connA1 = try pool.requestConnection().wait()
+        let connA1 = try await pool.requestConnection().get()
         XCTAssert(connA === connA1)
         pool.releaseConnection(connA1)
 
-        //keeping connection alive by using it and closing it
-        for _ in 1...3 {
+        // Keeping connection alive by using it and closing it
+        for _ in 0..<3 {
             try await Task.sleep(nanoseconds: UInt64(0.1 * Double(NSEC_PER_SEC)))
-            let connA2 = try pool.requestConnection().wait()
+            let connA2 = try await pool.requestConnection().get()
             XCTAssert(connA === connA2)
             pool.releaseConnection(connA2)
         }
-        
+
         pool.releaseConnection(anotherConnection1)
         pool.releaseConnection(anotherConnection2)
 
-        try await Task.sleep(nanoseconds: UInt64(0.3 * Double(NSEC_PER_SEC)))
+        try await Task.sleep(nanoseconds: UInt64(0.35 * Double(NSEC_PER_SEC)))
```

Comment:
it was a little flaky, possibly because the debugger, so i changed it to 0.35. It would still work with 0.3 most of the time.